components/automate-ui/src/app/pages/team-details/team-details.component.ts

```diff
@@ -134,37 +126,52 @@ export class TeamDetailsComponent implements OnInit, OnDestroy {
       }
     });
 
-    combineLatest([
+    const team$ = combineLatest([
+      this.isIAMv2$,
       this.store.select(v1TeamFromRoute),
       this.store.select(v2TeamFromRoute)
     ]).pipe(
       takeUntil(this.isDestroyed),
-      map(([v1Team, v2Team]: [Team, Team]) => {
-        return this.isIAMv2 ? v2Team : v1Team;
-      }),
-      filter(identity)
-    ).subscribe((team: Team) => {
-        this.team = team;
-        this.updateNameForm.controls.name.setValue(this.team.name);
-        this.store.dispatch(new GetTeamUsers({ id: this.teamId }));
-        this.store.dispatch(new GetUsers());
-      if (this.isIAMv2) {
-          this.store.dispatch(new GetProjects());
+      filter(([isV2, _v1TeamFromRoute, _v2TeamFromRoute]) => isV2 !== null),
+      map(([isV2, v1Team, v2Team]) => {
+        if (isV2) {
+          return v2Team;
+        } else {
+          return v1Team;
         }
-      });
+      }));
+
+    combineLatest([this.isIAMv2$, team$]).pipe(
```

Comment:
When the version and the team are found when can set the `this.teamId`, `this.team`, and request the TeamUsers.